### ISSUE TITLE
Fix useEffect usage in Discount

### ISSUE THREAD

**Issue by jmbrunskill** [msupply-foundation/open-msupply#4750]
I think we would need to fix the dependencies to useEffect in discount, although kind of hard to judge if that useEffect will run more then once, and ideally it's all should be done with 'await' rather then other hooks and dependencies. But that's for another issue/PR

_Originally posted by @andreievg in https://github.com/msupply-foundation/open-msupply/pull/4749#pullrequestreview-2284459562_

**Comment by andreievg** [comment]
Also would prefer if defaultPricing passed to createDraftStockOutLineFromStockLine is not optional https://github.com/msupply-foundation/open-msupply/blob/e98a73c8b2db764928ad11e0c519b83a2efbe11f/client/packages/invoices/src/StockOut/utils.tsx#L49-L62

**Comment by andreievg** [comment]
Also back end to return non optional fields ItemPriceNode

**Comment by andreievg** [comment]
@jmbrunskill I think this was fixed in the clean up issue/pr ?

**Comment by jmbrunskill** [comment]
> @jmbrunskill I think this was fixed in the clean up issue/pr ?

Yes, although it's still an optional, so I wasn't sure if you wanted to keep this open. The useEffect should be correct now...

**Comment by jmbrunskill** [comment]
Closing — this is implementation work inside the current front end, and the rewrite replaces that codebase rather than carrying it forward. A React hooks cleanup in the current front end: fixing a useEffect dependency array in the Discount component and making default pricing non-optional.

If we've got this wrong, just reopen it!

<sub>Part of the Sep 2026 backlog review (698 open issues), labelled `x. closed - Sep 2026 review` so the whole set stays findable and reversible. The review used automated code analysis — mistakes are possible.</sub>
